### PR TITLE
Add taskProvider bindings

### PR DESCRIPTION
**Pull request by Khady**
I want to write a task provider for dune. As a first step here are some bindings to the task provider api.
https://code.visualstudio.com/api/references/vscode-api#TaskProvider

In Task, I don't know what Workspace and Global refer to. So I only handle a workspace folder for now. But I still use a variant and bs.unwrap to mark that the api is incomplete here.
https://code.visualstudio.com/api/references/vscode-api#Task

Also I didn't bind the custom execution. Only shell and process execution are supported
https://code.visualstudio.com/api/references/vscode-api#CustomExecution

If the bindings can get merged I will then provide a second PR with the dune provider itself.

### CODE REVIEW

File: src/bindings/Vscode.ml [hunk 1]
```diff
@@ -133,6 +133,14 @@ module Workspace = struct
   external getConfiguration : string -> WorkspaceConfiguration.t
     = "getConfiguration"
     [@@bs.module "vscode"] [@@bs.scope "workspace"]
+
+  external findFiles :
+       inc:string
+    -> excl:string option
+    -> int option
```
Comment (rgrinberg) [inline src/bindings/Vscode.ml:140]: Is it possible to add a label to the int argument? It's not so easy to understand the purpose of it just by reading the signature.

File: src/bindings/Vscode.ml [hunk 2]
```diff
@@ -407,3 +415,83 @@ module LanguageClient = struct
     let open Promise.O in
     onReady t >>| fun () -> t.initializeResult
 end
+
+module ShellExecution = struct
+  type shellQuotingOptions
```
Comment (rgrinberg) [inline src/bindings/Vscode.ml:420]: How do we create values of this type?

File: src/bindings/Vscode.mli [hunk 3]
```diff
@@ -306,3 +313,74 @@ module LanguageClient : sig
...
+
+  type taskDefinition = { type_ : string [@bs.as "type"] }
+
+  val make :
+       taskDefinition:taskDefinition
+    -> scope:[ `Folder of Folder.t ]
+    -> name:string
+    -> source:string
+    -> execution:[ `Shell of ShellExecution.t | `Process of ProcessExecution.t ]
+    -> problemMatchers:string list option
```
Comment (rgrinberg) [inline src/bindings/Vscode.mli:368]: Is there a difference between `Some []` and `None` here?

File: src/bindings/Vscode.ml (hunk 2 above)
Comment (Khady) [reply src/bindings/Vscode.ml:420]: for now we don't, I didn't bind this part

File: src/bindings/Vscode.mli (hunk 3 above)
Comment (Khady) [reply src/bindings/Vscode.mli:368]: I think it should be the same, will remove the option

**Comment by Khady** [comment]
Sorry I was sure that I pushed my work last week. Should be good now

**Comment by rgrinberg** [comment]
Thanks!

**Pull request merged**